cpp/src/decisiontree/decisiontree_impl.cuh

```diff
@@ -177,34 +178,37 @@ void DecisionTreeBase<T, L>::plant(
          "Shared memory per block limit %zd , requested %zd \n", max_shared_mem,
          shmem_used);
 
-  for (int i = 0; i < MAXSTREAMS; i++) {
-    if (in_tempmem != nullptr) {
-      tempmem[i] = in_tempmem;
-    } else {
-      tempmem[i] = std::make_shared<TemporaryMemory<T, L>>(
-        handle, n_sampled_rows, ncols, MAXSTREAMS, unique_labels, n_bins,
-        split_algo);
-      quantile_per_tree = true;
-    }
-    if (split_algo == SPLIT_ALGO::GLOBAL_QUANTILE && quantile_per_tree) {
-      preprocess_quantile(data, rowids, n_sampled_rows, ncols, dinfo.NLocalrows,
-                          n_bins, tempmem[i]);
-    }
-    CUDA_CHECK(cudaStreamSynchronize(
-      tempmem[i]->stream));  // added to ensure accurate measurement
+  if (in_tempmem != nullptr) {
+    tempmem = in_tempmem;
+  } else {
+    tempmem = std::make_shared<TemporaryMemory<T, L>>(
+      handle, nrows, ncols, MAXSTREAMS, unique_labels, n_bins, split_algo,
+      maxdepth);
+    quantile_per_tree = true;
   }
+  if (split_algo == SPLIT_ALGO::GLOBAL_QUANTILE && quantile_per_tree) {
+    preprocess_quantile(data, rowids, n_sampled_rows, ncols, dinfo.NLocalrows,
+                        n_bins, tempmem);
+  }
+  CUDA_CHECK(cudaStreamSynchronize(
+    tempmem->stream));  // added to ensure accurate measurement
+
   prepare_time = prepare_fit_timer.getElapsedSeconds();
 
-  total_temp_mem = tempmem[0]->totalmem;
+  total_temp_mem = tempmem->totalmem;
   total_temp_mem *= MAXSTREAMS;
   MetricInfo<T> split_info;
   MLCommon::TimerCPU timer;
-  root = grow_tree(data, colper, labels, 0, rowids, n_sampled_rows, split_info);
+  if (split_algo == SPLIT_ALGO::GLOBAL_QUANTILE) {
```

Comment:
So this means quantile-based split algo only supports deep trees version?

Reply:
yes. for now. quantile is default algorithm.

Reply:
Got it. Thanks

Reply:
Once this PR is in. I will also add minmax algorithm. so after that the dependency on old code of Decision Tree will be removed.

Reply:
sounds like a good plan. Thanks Vishal.